### DISCUSSION TITLE
MultiStreamAggregate does not invoke Apply() or ProjectEvent lambda

### DISCUSSION THREAD

**Discussion by agross** [JasperFx/marten#2503] [General] [answered]
Hello!

I've tried to use `MultiStreamAggregation` with Marten 6.0.0-alpha.7.

```csharp
public class UserBookingByWorkplace
{
  [Identity]
  public string UserId { get; set; }
}

public class Works : EventProjection
{
  public Works()
  {
    Project<BookingAdded>((e, ops) =>
    {

    });
  }
}

public class Broken : MultiStreamAggregation<UserBookingByWorkplace, string>
{
  public Broken()
  {
    Identity<BookingAdded>(x => x.UserId);

    ProjectEvent<BookingAdded>((session, workplace, @event) => { });
  }
}
```

I've made sure a `BookingAdded` is added to a stream by my application, but only the first projection is invoked. Both are added to the `StoreOptions`.

```csharp
options.Projections.Add<Broken>();
options.Projections.Add<Works>();
```

I set a breakpoint in EventGraph.cs as decompiled by Rider:

```
IProjection[] projectionArray = eventGraph._inlineProjections.Value;
for (int index = 0; index < projectionArray.Length; ++index)
  await projectionArray[index].ApplyAsync((IDocumentOperations) session, (IReadOnlyList<StreamAction>) session.WorkTracker.Streams.ToList<StreamAction>(), token).ConfigureAwait(false);
```

The `projectionArray` does not contain the generated class for the `Broken` projection.

<img width="546" alt="2023-02-12_14-32-17" src="https://user-images.githubusercontent.com/16245/218314090-5f6ca9d2-1219-456c-9f27-6bc5bbe7d92e.png">

**Comment by jeremydmiller** [comment, marked as answer]
I'm going to convert this one to a discussion because I'm gonna go out on a limb and say this is a usage thing,

The Multi-stream aggregation is "async" by default. Are you running the async daemon?

**Comment by agross** [comment]
> Are you running the async daemon?

No, there's no setup regarding the Async Daemon so I guess it's off by default and I was not aware this is required. The [docs](https://martendb.io/events/projections/multi-stream-projections.html#multi-stream-aggregations) don't mention that it is required.

**Reply by jeremydmiller** [reply]
The documentation does state that the async daemon has to be explicitly added to the system, but we'd happily take a PR or just a suggestion on the docs to make that more clear.

https://martendb.io/events/projections/async-daemon.html

It's not on by default because it's potentially harmful in some use cases (Blazor, Azure Functions I think, Orleans (?)) and you'd never need it if you didn't have any async projections.

**Comment by agross** [comment]
#2508
